pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1438,9 +1459,14 @@ public boolean hasMessageAvailable() throws PulsarClientException {
         }
 
         AtomicLong opTimeoutMs = new AtomicLong(client.getConfiguration().getOperationTimeoutMs());
-        Backoff backoff = new Backoff(100, TimeUnit.MILLISECONDS,
-            opTimeoutMs.get() * 2, TimeUnit.MILLISECONDS,
-            0 , TimeUnit.MILLISECONDS);
+        Backoff backoff = new BackoffBuilder()
+                .setInitialTime(100, TimeUnit.MILLISECONDS)
```

Comment:
I think the user defined time should just override the initial time here.

Reply:
Just a question: where would the user defined time be found then? I basically copied the ```100``` milliseconds parameter since it was hardcoded before I made the change.

Reply:
@merlimat are you thinking something like

```
.setInitialTime(backoffIntervalNanos, TimeUnit.NANOS)
```

Reply:
@merlimat If this is the case, to me its kind of awkward. The unit for initial time is ```TimeUnit.MILLISECONDS```,  in contrast, backoffIntervalNanos is ```TimeUnit.NANOS```. So we probably have to convert these values back and forth to yield the backoff intervals with the right units, but at that point, things become a lot more convoluted than neccesary.